### PR TITLE
Allow commutative multiplication in Variant

### PR DESCRIPTION
**Pull request by vnen**
Also allow quaternions to be multiplied by integers.

### CODE REVIEW

File: core/math/vector2.h [hunk 1]
```diff
@@ -150,7 +150,19 @@ _FORCE_INLINE_ Vector2 Vector2::plane_project(real_t p_d, const Vector2 &p_vec)
...
 }
 
-_FORCE_INLINE_ Vector2 operator*(real_t p_scalar, const Vector2 &p_vec) {
+_FORCE_INLINE_ Vector2 operator*(float p_scalar, const Vector2 &p_vec) {
+	return p_vec * p_scalar;
+}
+
+_FORCE_INLINE_ Vector2 operator*(double p_scalar, const Vector2 &p_vec) {
+	return p_vec * p_scalar;
+}
```
Comment (akien-mga) [inline core/math/vector2.h:159]: Why does this need to be split in `float` and `double` variants?

Comment (Calinou) [reply core/math/vector2.h:159]: I suppose this is done so the multiplication can be done both using a float and double scalar, without incurring a performance loss due to automatic type casting.

Comment (vnen) [reply core/math/vector2.h:159]: This is to resolve ambiguity for the compiler.

Comment (akien-mga) [reply core/math/vector2.h:159]: But it's not needed for `Color` and `Quat`?

Comment (vnen) [reply core/math/vector2.h:159]: No. The problem here is that Vector2 and Vector2i are interchangeable. So the compiler don't know if it should use the operator for Vector2 or the one for Vector2i. I'm not sure why it's not a problem with Vector3 though.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**